packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/InAppPurchaseStoreKit2.swift

```diff
@@ -103,6 +103,30 @@ extension InAppPurchasePlugin: InAppPurchase2API {
     }
   }
 
+  func restorePurchases(completion: @escaping (Result<Void, Error>) -> Void) {
+    Task {
+      @MainActor in
+      do {
+        for await completedPurchase in Transaction.currentEntitlements {
+          switch completedPurchase {
+          case .verified(let purchase):
+            self.sendTransactionUpdate(
+              transaction: purchase, receipt: "\(completedPurchase.jwsRepresentation)")
+          case .unverified(_, _):
```

Comment:
should we early return if one of the entitlements fails

Reply:
i changed it so that we keep restoring even it fails, then complete at the very end with all the failures.